Tests/New-CEventLog.Tests.ps1

```diff
@@ -0,0 +1,79 @@
+
+#Requires -RunAsAdministrator
+#Requires -Version 5.1
+Set-StrictMode -Version 'Latest'
+
+BeforeAll {
+    Set-StrictMode -Version 'Latest'
+
+    & (Join-Path -Path $PSScriptRoot -ChildPath 'Initialize-Test.ps1' -Resolve)
+
+    $script:logName = 'New-CEventLog.Test'
+
+    Uninstall-CEventLog -LogName $script:logName
+
+    function GivenEventSource
+    {
+        param (
+            [String[]] $Source
+        )
+
+        foreach ($s in $Source)
+        {
+            [void] $script:source.Add($s)
+        }
+    }
+
+    function WhenCreatingEventLog
+    {
+        New-CEventLog -LogName $script:logName -Source $script:source -ErrorAction 'SilentlyContinue'
+    }
+
+    function ThenEventLogIsCreated
+    {
+        $log = [Diagnostics.EventLog]::New($script:logName)
+        $log | Should -Not -BeNullOrEmpty
+    }
+
+    function ThenEventLogHasSource
+    {
+        foreach ($s in $script:source)
+        {
+            [Diagnostics.EventLog]::SourceExists($s, $script:logName) | Should -Be $true
+        }
+    }
+
+    function ThenError
+    {
+        $Global:Error[0].Exception.Message | Should -Match 'The source exists'
+    }
+}
+
+Describe 'New-CEventLog' {
+    BeforeEach {
+        $script:source = [Collections.ArrayList]::New()
+        $Global:Error.Clear()
+    }
+
+    It 'should create an event log with the given source' {
+        GivenEventSource -Source 'Carbon.Windows'

```

Comment:
Lint: avoid using `Given` to set parameters for the function being tested. Instead, pass args for the function being tested to the `When` function, e.g.

```powershell
WhenCreatingEventLog -WithArgs @{ 'Source' = 'Carbon.Windows' }
```